### PR TITLE
created github pages directory

### PR DESCRIPTION
**Pull request by aidanm3341**
added a site directory with documentation of the Message Broker.

also create the relevant GitHub Actions to deploy it and keep it up to date with dependabot

### CODE REVIEW

**Comment by mimiflynn** [comment]
It looks like the site build is failing due to a missing `/site/package-lock.json`. If you run the build locally and commit the resulting `package-lock.json` it should resolve this issue.

File: site/gatsby-config.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const { plugins } = require('./src/config/base-gatsby-plugins');
+
+module.exports = {
+  siteMetadata: {
+    title: `Morgan Stanley Open Source Software`,
+    description: `Morgan Stanley Open Source Software`,
+    siteUrl: 'http://opensource.morganstanley.com',
```
Comment (mimiflynn) [inline site/gatsby-config.js:7]:
```suggestion
    siteUrl: 'http://opensource.morganstanley.com/message-broker',
```

File: site/gatsby-config.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+    documentationUrl: false,
+    //  documentationUrl: url-of.documentation.site,
+  },
+  pathPrefix: `/`, // include subdirectory
```
Comment (mimiflynn) [inline site/gatsby-config.js:11]:
```suggestion
  pathPrefix: `/message-broker`, // include subdirectory
```

File: site/gatsby-config.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
+const { plugins } = require('./src/config/base-gatsby-plugins');
+
+module.exports = {
+  siteMetadata: {
```
Comment (mimiflynn) [inline site/gatsby-config.js:4]: Fill our `title` and `description` for this project similar to the [ComposeUI](https://github.com/morganstanley/ComposeUI/blob/b3a21e454f785a8e4825f6f39e810876c1c002c2/site/gatsby-config.js) config.

File: site/content/news/hello-world.mdx
Comment (mimiflynn) [inline site/content/news/hello-world.mdx:1]: Probably should remove or replace the content of this page.

**Review by mimiflynn — CHANGES_REQUESTED** [review]
Looks good! Almost there!

File: site/content/news/hello-world.mdx
Comment (aidanm3341) [reply site/content/news/hello-world.mdx:1]: When I removed this, I got multiple of the following error: 'Cannot query field "date" on type "MdxFrontmatter".' when I try and run the site. I managed to fix it just by adding a dummy date to one of the other files, but it looks like the graphql queries requires that there is a date defined _somewhere_ in the site which is a little awkward.

**Pull request merged**